.travis.yml

```diff
@@ -37,6 +37,12 @@ jobs:
       sudo: required
       env: TOXENV=py37-ansi2html
 
+    -
+      python: 3.8
+      dist: xenial
+      sudo: required
+      env: TOXENV=devel
+
```

Comment:
I would like for this job to only run on `master` branch.

```
branches:
  only:
  - master
```

We've setup a cron-job to run daily.

I don't want this job to run on PRs, because I don't think that burden (if it fails) should land on the PR-creator. Especially since a failure can be unrelated to the changes in the PR.

That's for us maintainers to deal with.

Reply:
@BeyondEvil your personal preference makes it impossible to validate that this change does anything. I do not trust myself to merge stuff in the hope they will run correctly later.

Still, I updated travis config to allow failure of devel.

Reply:
Not sure what that means... 🤔

Why was running this only on master an issue?

I expect that when it fails, people (admins, members, etc.) are notified?

However, that raises an interesting question. What I would like to see is that if the CRON job fails, an issue is automatically created.

Moving to GA should definitely solve that, yes?

I’m sure such an integration ex

Reply:
Oh, I think I see what you’re saying (I would’ve worded it differently tho).

No worries, I have a PR in the works to add and update to python 3.8 everywhere. I can do it in that.

Reply:
Nope, cron execution does not create tickets neither on travis or github-actions. In theory it can be implemented but it has some security risks as it requires tokens with write access to github. Also there are some other practical reasons why I never seen it implemented: it is too easy to get the repo spammed with broken build reports.

That is why everyone I seen relies on email send from CI on failure. If I am correct they do email cores by default.

Regarding "running only on master" question, let me explain. Look at travis checks on current change, you do see "devel" among them, right? Clearly we do want to see that "tox -e devel" runs on current change? If we configure it to run only on master branch, we would not be able to see the result of devel run here anymore, making impossible to test any fix related to devel, as we would be driving blindly, we would know the result only after the merge.

Reply:
> Nope, cron execution does not create tickets neither on travis or github-actions.

I know that. What I meant was that someone might have created an integration you could install/activate.

> In theory it can be implemented but it has some security risks as it requires tokens with write access to github. 

Well sure, but the GH tokens can be tightly scoped. Already today, the world can create an issue in the project.

> Also there are some other practical reasons why I never seen it implemented: it is too easy to get the repo spammed with broken build reports.

Well sure, but that spamming occurs in email-boxes today.